### PR TITLE
Do not formatting nonexist cm and secret

### PR DESCRIPTION
**Pull request by tomahawk28**
To resolve an issue with e2e test on app-operator, we need to wrap around cm, secret with if statements that do not need to create.

### CODE REVIEW

File: pkg/chartvalues/apiextensions_app_e2e_test.go [hunk 1]
```diff
@@ -131,6 +135,54 @@ secrets:
...
+    description: "giantswarm catalog"
+    logoUrl: "http://giantswarm.com/catalog-logo.png"
+    storage:
+      type: "helm"
+      url: "https://giantswarm.github.com/giantswarm-catalog/"
+
+appOperator:
+  version: "1.0.0"
+
+
```
Comment (tomahawk28) [inline pkg/chartvalues/apiextensions_app_e2e_test.go:104]: Those are ugly white space, but can't find a way to formatting well with go templates :(

Comment (rossf7) [reply pkg/chartvalues/apiextensions_app_e2e_test.go:104]: Yes these are hard to manage. Could we have a constant so the UT code is cleaner?

File: pkg/chartvalues/apiextensions_app_e2e_test.go [hunk 2]
```diff
@@ -59,6 +59,10 @@ func newAPIExtensionsAppE2EConfigFromFilled(modifyFunc func(*APIExtensionsAppE2E
 	return c
 }
 
+func removeConfigMapAndSecret(v *APIExtensionsAppE2EConfig) {
+	v.App.Config = APIExtensionsAppE2EConfigAppConfig{}
+}
```
Comment (rossf7) [inline pkg/chartvalues/apiextensions_app_e2e_test.go]: I think this makes sense and it avoids defining another struct which is big.

**Review by rossf7 — APPROVED** [review]
LGTM just a suggestion on the UT.

File: pkg/chartvalues/apiextensions_app_e2e_test.go (hunk 1 above)
Comment (tomahawk28) [reply pkg/chartvalues/apiextensions_app_e2e_test.go:104]: Sure, I move all of the expected values as const.

**Pull request merged**